### PR TITLE
python3Packages.mutmut: init at 2.2.0

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mutmut/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  postPatch = ''
+    sed -i 's/junit-xml==1.8/junit-xml==1.9/' requirements.txt
+  '';
+
+  disabled = pythonOlder "3.7";
+
+  doCheck = false;
+
+  propagatedBuildInputs = [ click glob2 parso pony junit-xml ];
```
Comment (fabaff) [inline pkgs/development/python-modules/mutmut/default.nix:32]: I guess that this requires #155230 for Python 3.10 too.

File: pkgs/development/python-modules/mutmut/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "boxed";
+    rev = version;
+    hash = "sha256-G+OL/9km2iUeZ1QCpU73CIWVWMexcs3r9RdCnAsESnY=";
+  };
+
+  postPatch = ''
+    sed -i 's/junit-xml==1.8/junit-xml==1.9/' requirements.txt
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/mutmut/default.nix:25]: Is there any reason to prefer this instead of `substituteInPlace`?

Comment (Synthetica9) [reply pkgs/development/python-modules/mutmut/default.nix:25]: No, I initially wrote this as a personal thing before I decided to upstream it. I'll change it

**Comment by Synthetica9** [comment]
Gonna merge this if no-one else has any objections.

**Pull request merged**